From c5291ab1ca4d4d7a041874a66b1c05b7d5c02ccb Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Wed, 26 Jan 2011 23:40:25 +0000
Subject: [PATCH] CODEC-112 Base64.encodeBase64(byte[] binaryData, boolean
 isChunked, boolean urlSafe, int maxResultSize) throws IAE for valid
 maxResultSize if isChunked is false

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/codec/trunk@1063922 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/commons/codec/binary/Base64.java     | 2 +-
 src/test/org/apache/commons/codec/binary/Base64Test.java | 8 +++++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/java/org/apache/commons/codec/binary/Base64.java b/src/java/org/apache/commons/codec/binary/Base64.java
index 2cd0163b06..e81b7e9138 100644
--- a/src/java/org/apache/commons/codec/binary/Base64.java
+++ b/src/java/org/apache/commons/codec/binary/Base64.java
@@ -824,7 +824,7 @@ public static byte[] encodeBase64(byte[] binaryData, boolean isChunked, boolean
             return binaryData;
         }
 
-        long len = getEncodeLength(binaryData, MIME_CHUNK_SIZE, CHUNK_SEPARATOR);
+        long len = getEncodeLength(binaryData, isChunked ? MIME_CHUNK_SIZE : 0, CHUNK_SEPARATOR);
         if (len > maxResultSize) {
             throw new IllegalArgumentException("Input array too big, the output array would be bigger (" +
                 len +
diff --git a/src/test/org/apache/commons/codec/binary/Base64Test.java b/src/test/org/apache/commons/codec/binary/Base64Test.java
index 428b54a585..0da85d05be 100644
--- a/src/test/org/apache/commons/codec/binary/Base64Test.java
+++ b/src/test/org/apache/commons/codec/binary/Base64Test.java
@@ -336,13 +336,19 @@ public void testEncodeOverMaxSize() throws Exception {
         testEncodeOverMaxSize(1);
         testEncodeOverMaxSize(2);
     }
+    
+    public void testCodec112() { // size calculation assumes always chunked
+        byte[] in = new byte[] {0};
+        byte[] out=Base64.encodeBase64(in);
+        Base64.encodeBase64(in, false, false, out.length);
+    }
 
     private void testEncodeOverMaxSize(int maxSize) throws Exception {
         try {
             Base64.encodeBase64(Base64TestData.DECODED, true, false, maxSize);
             fail("Expected " + IllegalArgumentException.class.getName());
         } catch (IllegalArgumentException e) {
-            // Expceted
+            // Expected
         }
     }
 
